angular2-twitter/public/src/app/linear-graph/linear-graph.component.ts

```diff
@@ -0,0 +1,122 @@
+import { Component, OnInit, Input, OnChanges } from '@angular/core';
+import * as d3 from 'd3';
+
+import { GraphService } from '../graph.service';
+import { Dimensions, LinearData } from '../app.classes';
+
+@Component({
+  selector: 'app-linear-graph',
+  templateUrl: './linear-graph.component.html',
+  styleUrls: ['./linear-graph.component.css']
+})
+export class LinearGraphComponent implements OnInit {
+
+  dimensions: Dimensions;
+
+  constructor(private graphService: GraphService) { 
+    this.dimensions = {
+      width: 800,
+      height: 400,
+      margin: {
+        top: 20, 
+        right: 30, 
+        bottom: 30, 
+        left: 80
+      }
+    }
+  }
+
+  @Input() linearData: LinearData[];
+
+  ngOnInit() { }
```

Comment:
jak nie używasz to nie deklaruj, wiem, że ng cli dodaje to automatycznie, ale nie jest to dobra praktyka